src/System.Data.SqlClient/src/Common/Interop/Unix/System.Net.Security.Native/Interop.NetSecurityNative.Extended.cs

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class NetSecurityNative
+    {
+        [DllImport(Interop.Libraries.NetSecurityNative, EntryPoint = "NetSecurityNative_EnsureGssInitialized")]
```

Comment:
Typically we don't think of the native shims as public API.  Since you're depending on this from a NuGet package it's now effectively public API that can never change.  Has that been discussed with the owners of that component? @dotnet/ncl @VSadov 

Also, I don't see this export at all in 3.1: https://github.com/dotnet/corefx/blob/release/3.1/src/Native/Unix/System.Net.Security.Native/pal_gssapi.c

Do you need to port https://github.com/dotnet/runtime/pull/55037 as well?

Reply:
This is very old issue: https://github.com/dotnet/runtime/issues/24115

The networking team promised to maintain compatibility of System.Net.Security.Native shims for SqlClient as short-term fix. That was in 2017. We are still waiting on the long-term fix.

Reply:
This is an effort to keep System.Data.SqlClient functional against .NET 6+. Fixing the internals calls by both SqlClients (dotnet/runtime#24115) is blocked by dotnet/runtime#29270, which is the public API replacement for those calls.